src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery.Test/ScenarioTests/A2A/AsrA2ATestsBase.cs

```diff
@@ -0,0 +1,242 @@
+        public RecoveryServicesClient RecoveryServicesMgmtClient { get; private set; }
+        public SiteRecoveryManagementClient SiteRecoveryMgmtClient { get; private set; }
+
+        public SubscriptionClient SubscriptionClient { get; private set; }
+
+        public ComputeManagementClient ComputeManagementClient { get; private set; }
+        public Microsoft.Azure.Management.ResourceManager.ResourceManagementClient ResourceManagementRestClient { get; private set; }
+
+        public void RunPowerShellTest(
+             XunitTracingInterceptor logger,
+             string scenario,
+            params string[] scripts)
+        {
+            var callingClassType = TestUtilities.GetCallingClass(2);
+            var mockName = TestUtilities.GetCurrentMethodName(2);
+
+            this.RunPsTestWorkflow(
+                scenario,
+                () => scripts,
+
+                // no custom initializer
+                null,
+
+                // no custom cleanup 
+                null,
+                callingClassType,
+                mockName);
+        }
+
+        public void RunPsTestWorkflow(
+            string scenario,
+            Func<string[]> scriptBuilder,
+            Action<CSMTestEnvironmentFactory> initialize,
+            Action cleanup,
+            string callingClassType,
+            string mockName)
+        {
+            var providers = new Dictionary<string, string>();
+            providers.Add("Microsoft.Resources", null);
+            providers.Add("Microsoft.Features", null);
+            providers.Add("Microsoft.Authorization", null);
+
+            var providersToIgnore = new Dictionary<string, string>();
+            providersToIgnore.Add("Microsoft.Azure.Management.Resources.ResourceManagementClient", "2016-02-01");
+            providersToIgnore.Add("Microsoft.Azure.Management.ResourceManager.ResourceManagementClient", "2017-05-10");
+            providersToIgnore.Add("Microsoft.Azure.Management.Internal.Resources.ResourceManagementClient", "2016-09-01");
+
+            HttpMockServer.Matcher = new PermissiveRecordMatcherWithApiExclusion(
+                true,
+                providers,
+                providersToIgnore);
+
+            HttpMockServer.RecordsDirectory = Path.Combine(
+                AppDomain.CurrentDomain.BaseDirectory,
+                "SessionRecords");
+
+            using (var context = RestTestFramework.MockContext.Start(
+                callingClassType,
+                mockName))
+            {
+                this.csmTestFactory = new CSMTestEnvironmentFactory();
+
+                if (initialize != null)
+                {
+                    initialize.Invoke(this.csmTestFactory);
+                }
+
+                this.SetupManagementClients(context);
+
+                this.helper.SetupEnvironment(AzureModule.AzureResourceManager);
+
+                var rmProfileModule = this.helper.RMProfileModule;
+                var modules = new List<string>();
```

Comment:
Why not simply declare an array with the values already initialized? No need to dynamically add values and then convert it to an array.